spec/models/multi_application_account_spec.rb

```diff
@@ -0,0 +1,239 @@
+      end
+    end
+
+    context "with host given" do
+      context "when account exists" do
+        it "loads the existing account" do
+          expect(MultiApplicationsAccount.from_omniauth(auth, "example.test")).to eql(account)
+        end
+
+        describe "the updated account" do
+          before do
+            MultiApplicationsAccount.from_omniauth(auth, "example.test")
+            account.reload
+          end
+
+          it_behaves_like "it takes attributes from auth"
+        end
+      end
+
+      context "when account doesn't exist" do
+        it "creates new account" do
+          expect {
+            MultiApplicationsAccount.from_omniauth(auth, "example3.test")
+          }.to change { MultiApplicationsAccount.count }.by(1)
+        end
+
+        describe "the newly created account" do
+          let!(:account) { MultiApplicationsAccount.from_omniauth(auth, "example3.test") }
+
+          it "sets synced_id and provider from auth as well as host" do
+            expect(account.synced_id).to eq 123
+            expect(account.provider).to eq "bookingsync"
+            expect(account.host).to eq "example3.test"
+          end
+
+          it_behaves_like "it takes attributes from auth"
+        end
+      end
+    end
+  end
+
+  describe ".find_by_host_and_synced_id" do
+    let!(:account_1) { MultiApplicationsAccount.create!(synced_id: 1, host: "example.test") }
+    let!(:account_2) { MultiApplicationsAccount.create!(synced_id: 2, host: "example.test") }
+    let!(:account_3) { MultiApplicationsAccount.create!(synced_id: 1, host: "example2.test") }
+    let!(:account_4) { MultiApplicationsAccount.create!(synced_id: 2, host: "example2.test") }
+
+    it "returns the right account" do
+      expect(MultiApplicationsAccount.find_by_host_and_synced_id("example2.test", 1)).to eq account_3
+    end
+  end
+
+  describe "#token" do
+
+    let!(:account) do
+      MultiApplicationsAccount.create!(
+        synced_id: 123, host: "example.test",
+        oauth_access_token: "token", oauth_refresh_token: "refresh_token",
+        oauth_expires_at: expires_at
+      )
+    end
+    let!(:application) { Application.create!(host: "example.test", client_id: "123", client_secret: "456") }
+
+    context "when the stored token is fresh" do
+      let(:expires_at) { 1.day.from_now.to_i }
+
+      it "returns the token" do
+        expect(account.token).to be_a OAuth2::AccessToken
+        expect(account.token.token).to eq "token"
+      end
+    end
+
+    context "when the stored token is expired" do
+      around do |test_case|
+        # comparing rails version, the use_transactional_fixtures only works pre 5
+        if Rails::VERSION::MAJOR >= 5
+          orinal_setup = self.use_transactional_tests
+          self.use_transactional_tests = false
+          test_case.run
+          self.use_transactional_tests = orinal_setup
+        else
+          orinal_setup = self.use_transactional_fixtures
```

Comment:
those 2 conditionals are the same

Reply:
Not sure if you misread the methods or if I didn't understood your review:

First branch is `use_transactional_TESTS` and other branch is `use_transactional_FIXTURES`

Reply:
Oh, right, I saw the same suffix in both cases 😅